usecase/reservation/reservation.go

```diff
@@ -0,0 +1,36 @@
+package reservation
+
+import (
+	"room_app_back/domain/model"
+	"room_app_back/domain/repository/reservation"
+)
+
+type ReservationUsecase struct {
+	rr reservation.ReservationRepository
+}
+
+func NewReservationUsecase(rr reservation.ReservationRepository) *ReservationUsecase {
+	return &ReservationUsecase{
+		rr: rr,
+	}
+}
+
+func (ru *ReservationUsecase) Add(r model.Reservation) (*model.Reservation, error) {
```

Comment:
大体、関数の引数と返り値に指定するStructの型はポインタであることが多いです。

```suggestion
func (ru *ReservationUsecase) Add(r *model.Reservation) (*model.Reservation, error) {
```